analysis/analysis.py

```diff
@@ -282,15 +283,135 @@ def get_word_frequency_differences(student_data):
     return ordered_responses
 
 
-def run_analysis():
+def compute_median_view_time(student_data):
+    """
+     Given a list of student response dicts,
+    return the median time (across all users) spent reading the text
+
+    :param student_data: list, student response dicts
+    :return: float, median amount of time users spend reading the text
+    """
+    list_of_times = []
+    for row in student_data:
+        for view_time in row.get('views'):
+            list_of_times.append(view_time)
+    list_of_times.sort()
```

Comment:
you don't need to call `sort()` if the length of the list is 0, so this can go in the else branch below